### PR TITLE
Updated user manual to be aligned to debug specification version 1.0.0.

### PR DESCRIPTION
**Pull request by silabs-oysteink**
CSR scontext renamed to mscontext according to debug spec 1.0.0.

Signed-off-by: Oystein Knauserud <Oystein.Knauserud@silabs.com>

### CODE REVIEW

File: docs/user_manual/source/control_status_registers.rst [hunk 1]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
 
 Reset Value: 0x0000_0000
 
-Accessible in Debug Mode or M-Mode.
+Accessible in Debug Mode or M-Mode, depending on **TDATA1.dmode**.
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:721]: Is this really correct?

File: docs/user_manual/source/control_status_registers.rst [hunk 2]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
...
+If a value larger than the parameter ``DBG_NUM_TRIGGERS`` is written, the register will contain the value DBG_NUM_TRIGGERS - 1.
 
 +-------------+-----------+----------------------------------------------------------------------------------------+
 |   Bit #     |   Mode    |   Description                                                                          |
 +=============+===========+========================================================================================+
-| 31:0        | RO        | |corev| implements a single trigger, therefore this register will always read as zero  |
+|| 31:0       || RW       || |corev| implements 0 to 4 triggers based on the parameter DBG_NUM_TRIGGERS. Selects   |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:727]: What are the double || for?

File: docs/user_manual/source/control_status_registers.rst [hunk 3]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
...
-Accessible in Debug Mode or M-Mode.
-Since native triggers are not supported, writes to this register from M-Mode will be ignored.
+Accessible in Debug Mode or M-Mode, depending on **TDATA1.dmode**. The contents of the **data** field depends on the current
+value of the **type** field. See [RISC-V-DEBUG]_ for details regarding all trigger related CSRs.
 
-.. note::
++-------+-----------+----------------------------------------------------------------+
+| Bit#  | Mode      | Description                                                    |
++=======+===========+================================================================+
+|| 31:28|| RW       || **type:** 6 = Address match trigger type.                     |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:746]: What are the double || for? Same question for all modified tables.

File: docs/user_manual/source/control_status_registers.rst [hunk 4]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
...
++-------+-----------+----------------------------------------------------------------+
+
+.. _csr-etrigger:
+
+Exception Trigger (``etrigger``)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CSR Address: 0x7A1
+
+Reset Value: 0x5800_0000 (TBD)
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:820]: What part is TBD?

File: docs/user_manual/source/control_status_registers.rst [hunk 5]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
...
-| 0     | RO (0x0) | **load:** 0 = Load address / data matching not supported.        |
-+-------+----------+------------------------------------------------------------------+
+.. _csr-mcontrol6:
+
+Match Control Type 6 (``mcontrol6``)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CSR Address: 0x7A1
+
+Reset Value: 0x6000_1040 (TBD)
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:762]: What part is TBD?

File: docs/user_manual/source/control_status_registers.rst [hunk 6]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
...
 
-Trigger Data Register 1 (``tdata1``)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Trigger Data 1 (``tdata1``)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 CSR Address: 0x7A1
 
-Reset Value: 0x2800_1040
+Reset Value: 0x0000_0000 TBD
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:738]: What part is TBD?

File: docs/user_manual/source/control_status_registers.rst [hunk 7]
```diff
@@ -830,16 +896,16 @@ Trigger Info (``tinfo``)
...
-Reset Value: 0x0000_0004
+Reset Value: 0x0000_0060
 
 Detailed:
 
 +-------+----------+------------------------------------------------------------------+
 | Bit#  | Mode     | Description                                                      |
 +=======+==========+==================================================================+
-| 31:16 | RO (0x0) | 0                                                                |
+| 31:16 | RO (0x0) | Always returen 0                                                 |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:906]: returen -> returns

File: docs/user_manual/source/control_status_registers.rst [hunk 8]
```diff
@@ -716,74 +718,138 @@ CSR Address: 0x7A0
...
+
+Reset Value: 0x5800_0000 (TBD)
+
+Accessible in Debug Mode or M-Mode, depending on **TDATA1.dmode**.
+
+
++-------+-----------+----------------------------------------------------------------+
+| Bit#  | Mode      | Description                                                    |
++=======+===========+================================================================+
+| 31:28 | RW (0x5)  | **type:** 5 = Exception trigger.                               |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:828]: Remove (0x5)

File: docs/user_manual/source/control_status_registers.rst [hunk 9]
```diff
@@ -909,7 +988,11 @@ Detailed:
 +==========+===========+=================================================================================================+
 | 31:28    | RO (0x4)  | **xdebugver:** returns 4 - External debug support exists as it is described in this document.   |
```
Comment (Silabs-ArjanB) [inline docs/user_manual/source/control_status_registers.rst:989]: it is described in this document -> should point to RISC-V debug spec instead

**Review by Silabs-ArjanB — CHANGES_REQUESTED** [review]

File: docs/user_manual/source/debug.rst [hunk 10]
```diff
@@ -79,10 +81,12 @@ Core Debug Registers
...
 |corev| implements four core debug registers, namely :ref:`csr-dcsr`, :ref:`csr-dpc`, and two debug scratch registers. Access to these registers in non Debug Mode results in an illegal instruction.
 
-Several trigger registers are required to adhere to specification. The following are the most relevant: :ref:`csr-tselect`, :ref:`csr-tdata1`,  :ref:`csr-tdata2` and :ref:`csr-tinfo`
+Several trigger registers are included if ``DBG_NUM_TRIGGERS`` is set to a value greater than 0.
+The following are the most relevant: :ref:`csr-tselect`, :ref:`csr-tdata1`,  :ref:`csr-tdata2` and :ref:`csr-tinfo`
+If ``DBG_NUM_TRIGGERS`` is zero, access to the trigger registers will result in an illegal instruction exception.
 
-The TDATA1.DMODE is hardwired to a value of 1. In non Debug Mode,
-writes to Trigger registers are ignored and reads reflect CSR values.
+The TDATA1.DMODE controls write access permission to the trigger registers. A value of 0 permits machine mode to write to the trigger registers.
```
[the remark this answers is not in the document]
Comment (silabs-oysteink) [reply docs/user_manual/source/debug.rst:88]: You are right, dmode controls write permission to the currently selected triggers tdata registers.

File: docs/user_manual/source/control_status_registers.rst (hunk 1 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/control_status_registers.rst:721]: No it's not, I'll fix it.

File: docs/user_manual/source/control_status_registers.rst (hunk 9 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/control_status_registers.rst:989]: Done

File: docs/user_manual/source/control_status_registers.rst (hunk 8 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/control_status_registers.rst:828]: Done

File: docs/user_manual/source/control_status_registers.rst (hunk 7 above)
Comment (silabs-oysteink) [reply docs/user_manual/source/control_status_registers.rst:906]: Done

**Review by Silabs-ArjanB — APPROVED** [review]

**Pull request merged**